src/test/java/com/wojciechkocik/usage/service/CourseUsageServiceTest.java

```diff
@@ -0,0 +1,217 @@
+package com.wojciechkocik.usage.service;
+
+import com.wojciechkocik.usage.dto.CourseUsageCreate;
+import com.wojciechkocik.usage.dto.DailyUsageResponse;
+import com.wojciechkocik.usage.entity.CourseUsage;
+import com.wojciechkocik.usage.repository.CourseUsageRepository;
+import org.jfairy.Fairy;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.validation.ConstraintViolationException;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * @author Wojciech Kocik
+ * @since 22.03.2017
+ */
+@RunWith(SpringRunner.class)
+@SuppressWarnings("unused")
+@SpringBootTest
+public class CourseUsageServiceTest {
+
+    private Fairy fairy = Fairy.create();
+
+    @Autowired
+    private CourseUsageService courseUsageService;
+
+    @Autowired
+    private CourseUsageRepository courseUsageRepository;
+
+    @Before
+    public void setUp() {
+        courseUsageRepository.deleteAll();
+    }
+
+    @Test
+    public void createNewCourseUsage_dtoIsProperMappedToDatabase() {
+
+        //Arrange
+        String courseId = fairy.textProducer().randomString(10);
+        String userId = fairy.textProducer().randomString(10);
+        long timeSpent = 5000;
+        ZonedDateTime started = ZonedDateTime.now();
+
+        CourseUsageCreate courseUsageCreate = new CourseUsageCreate();
+        courseUsageCreate.setCourseId(courseId);
+        courseUsageCreate.setUserId(userId);
+        courseUsageCreate.setTimeSpent(timeSpent);
+        courseUsageCreate.setStarted(started);
+
+        CourseUsage courseUsageExpected = new CourseUsage();
+        courseUsageExpected.setTimeSpent(timeSpent);
+        courseUsageExpected.setUserId(userId);
+        courseUsageExpected.setCourseId(courseId);
+        courseUsageExpected.setStarted(started);
+
+        //Act
+        CourseUsage courseUsageActual = courseUsageService.createNewCourseUsage(courseUsageCreate);
+        courseUsageExpected.setId(courseUsageActual.getId());
```

Comment:
No, because it is expected object and I can't retieve it from db (I'm expecting that it will be in db). Or I don't understand you.